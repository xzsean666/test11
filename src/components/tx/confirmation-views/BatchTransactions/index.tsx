import BatchTxList from '@/components/batch/BatchSidebar/BatchTxList'
import { useDraftBatch } from '@/hooks/useDraftBatch'

function BatchTransactions() {
  const batchTxs = useDraftBatch()

  return <BatchTxList txItems={batchTxs} />
}

export default BatchTransactions
