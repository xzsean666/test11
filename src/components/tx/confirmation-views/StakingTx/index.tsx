import StrakingConfirmationTx from '@/features/stake/components/StakingConfirmationTx'
import type { StakingTxInfo } from '@safe-global/safe-gateway-typescript-sdk'
import type { NarrowConfirmationViewProps } from '../types'

export interface StakingTxProps extends NarrowConfirmationViewProps {
  txInfo: StakingTxInfo
}

function StakingTx({ txInfo }: StakingTxProps) {
  return <StrakingConfirmationTx order={txInfo} />
}

export default StakingTx
