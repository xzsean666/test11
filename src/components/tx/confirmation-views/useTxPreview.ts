import { Operation, getTxPreview } from '@safe-global/safe-gateway-typescript-sdk'
import type { SafeTransaction } from '@safe-global/safe-core-sdk-types'
import useAsync from '@/hooks/useAsync'
import useSafeInfo from '@/hooks/useSafeInfo'

const useTxPreview = (
  safeTxData?: {
    operation: SafeTransaction['data']['operation']
    data: SafeTransaction['data']['data']
    value: SafeTransaction['data']['value']
    to: SafeTransaction['data']['to']
  },
  customSafeAddress?: string,
  txId?: string,
) => {
  const skip = !!txId || !safeTxData
  const {
    safe: { chainId },
    safeAddress,
  } = useSafeInfo()
  const address = customSafeAddress ?? safeAddress
  const { operation = Operation.CALL, data = '', to, value } = safeTxData ?? {}

  return useAsync(() => {
    if (skip) return
    return getTxPreview(chainId, address, operation, data, to, value)
  }, [skip, chainId, address, operation, data, to, value])
}

export default useTxPreview
