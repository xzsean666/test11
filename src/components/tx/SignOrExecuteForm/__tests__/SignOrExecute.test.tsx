import SignOrExecute from '../index'
import { render } from '@/tests/test-utils'
import type { TransactionPreview } from '@safe-global/safe-gateway-typescript-sdk'
import type { SafeTxContextParams } from '@/components/tx-flow/SafeTxProvider'
import { SafeTxContext } from '@/components/tx-flow/SafeTxProvider'
import { createSafeTx } from '@/tests/builders/safeTx'
import * as useTxPreviewHooks from '@/components/tx/confirmation-views/useTxPreview'

let isSafeOwner = true
// mock useIsSafeOwner
jest.mock('@/hooks/useIsSafeOwner', () => ({
  __esModule: true,
  default: jest.fn(() => isSafeOwner),
}))

// Mock proposeTx
jest.mock('@/services/tx/proposeTransaction', () => ({
  __esModule: true,
  default: jest.fn(() => Promise.resolve({ txId: '123' })),
}))

jest.mock('@/hooks/useValidateTxData', () => ({
  __esModule: true,
  useValidateTxData: jest.fn().mockReturnValue([undefined, undefined, false]),
}))

describe('SignOrExecute', () => {
  beforeEach(() => {
    isSafeOwner = true
    jest.clearAllMocks()
  })

  it('should display a loading component', () => {
    const { container } = render(<SignOrExecute onSubmit={jest.fn()} isExecutable={true} />)

    expect(container).toMatchSnapshot()
  })

  it('should display a confirmation screen', async () => {
    jest.spyOn(useTxPreviewHooks, 'default').mockReturnValue([
      {
        txInfo: {},
        txData: {},
      } as TransactionPreview,
      undefined,
      false,
    ])

    const { container, getByTestId } = render(
      <SafeTxContext.Provider
        value={
          {
            safeTx: createSafeTx(),
          } as SafeTxContextParams
        }
      >
        <SignOrExecute onSubmit={jest.fn()} isExecutable={true} />
      </SafeTxContext.Provider>,
    )

    expect(getByTestId('sign-btn')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should display an error screen', async () => {
    jest
      .spyOn(useTxPreviewHooks, 'default')
      .mockReturnValue([undefined, new Error('This is a mock error message'), false])

    const { container } = render(
      <SafeTxContext.Provider
        value={
          {
            safeTx: createSafeTx(),
          } as SafeTxContextParams
        }
      >
        <SignOrExecute onSubmit={jest.fn()} isExecutable={true} />
      </SafeTxContext.Provider>,
    )

    expect(container.querySelector('sign-btn')).not.toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
