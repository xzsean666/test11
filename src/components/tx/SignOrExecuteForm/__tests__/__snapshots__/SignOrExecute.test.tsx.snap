// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOrExecute should display a confirmation screen 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1w4z6qv-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: none;"
  >
    <div
      class="MuiCardContent-root cardContent css-1lt5qva-MuiCardContent-root"
      data-testid="card-content"
    >
      <div
        class="MuiStack-root css-1sazv7p-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1gdgqta-MuiPaper-root-MuiAccordion-root"
            style="--Paper-shadow: none;"
          >
            <h3
              class="MuiAccordion-heading css-cy7rkm-MuiAccordion-heading"
            >
              <button
                aria-expanded="false"
                class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters accordion css-10sjung-MuiButtonBase-root-MuiAccordionSummary-root"
                data-testid="decoded-tx-summary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-1r0e0ir-MuiAccordionSummary-content"
                >
                  Advanced details
                  <span
                    class=""
                    data-mui-internal-clone-element="true"
                  >
                    <mock-icon
                      aria-hidden=""
                      class="MuiSvgIcon-root MuiSvgIcon-colorBorder MuiSvgIcon-fontSizeSmall css-17ibv3e-MuiSvgIcon-root"
                      focusable="false"
                    />
                  </span>
                  <div
                    class="MuiBox-root css-1rr4qq7"
                  />
                </span>
                <span
                  class="MuiAccordionSummary-expandIconWrapper css-1wqf3nl-MuiAccordionSummary-expandIconWrapper"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1dhtbeh-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </span>
              </button>
            </h3>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-cwrbtg-MuiCollapse-root"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
                >
                  <div
                    class="MuiAccordion-region"
                    role="region"
                  >
                    <div
                      class="MuiAccordionDetails-root css-w74p4c-MuiAccordionDetails-root"
                      data-testid="decoded-tx-details"
                    >
                      <div
                        class="MuiStack-root css-14g2pqt-MuiStack-root"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-111k8jw-MuiTypography-root"
                        >
                          Transaction data
                        </p>
                        <div
                          class="MuiGrid-root MuiGrid-container css-86qz5o-MuiGrid-root"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item css-1wehkfy-MuiGrid-root"
                            data-testid="tx-row-title"
                            style="word-break: break-word;"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-shf88x-MuiTypography-root"
                            >
                              to:
                            </p>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
                            data-testid="tx-data-row"
                          >
                            <div
                              class="container"
                            >
                              <div
                                class="MuiBox-root css-1lchl8k"
                              >
                                <div
                                  class="addressContainer"
                                >
                                  <div
                                    class="MuiBox-root css-b5p5gz"
                                  >
                                    <span
                                      aria-label="Copy to clipboard"
                                      class=""
                                      data-mui-internal-clone-element="true"
                                      style="cursor: pointer;"
                                    >
                                      <span>
                                        0x0000000000000000000000000000000000000000
                                      </span>
                                    </span>
                                  </div>
                                  <span
                                    aria-label="Copy to clipboard"
                                    class=""
                                    data-mui-internal-clone-element="true"
                                    style="cursor: pointer;"
                                  >
                                    <button
                                      aria-label="Copy to clipboard"
                                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-xyrzjn-MuiButtonBase-root-MuiIconButton-root"
                                      tabindex="0"
                                      type="button"
                                    >
                                      <mock-icon
                                        aria-hidden=""
                                        class="MuiSvgIcon-root MuiSvgIcon-colorBorder MuiSvgIcon-fontSizeSmall css-gvpe62-MuiSvgIcon-root"
                                        data-testid="copy-btn-icon"
                                        focusable="false"
                                      />
                                    </button>
                                  </span>
                                  <div
                                    class="MuiBox-root css-yjghm1"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-container css-86qz5o-MuiGrid-root"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item css-1wehkfy-MuiGrid-root"
                            data-testid="tx-row-title"
                            style="word-break: break-word;"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-shf88x-MuiTypography-root"
                            >
                              value:
                            </p>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
                            data-testid="tx-data-row"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-mxvcov-MuiTypography-root"
                            >
                              0x0
                            </p>
                          </div>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-container css-86qz5o-MuiGrid-root"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item css-1wehkfy-MuiGrid-root"
                            data-testid="tx-row-title"
                            style="word-break: break-word;"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-shf88x-MuiTypography-root"
                            >
                              data:
                            </p>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
                            data-testid="tx-data-row"
                          >
                            <div
                              class="encodedData MuiBox-root css-0"
                              data-testid="tx-hexData"
                            >
                              <span
                                aria-label="Copy to clipboard"
                                class=""
                                data-mui-internal-clone-element="true"
                                style="cursor: pointer;"
                              >
                                <button
                                  aria-label="Copy to clipboard"
                                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-xyrzjn-MuiButtonBase-root-MuiIconButton-root"
                                  tabindex="0"
                                  type="button"
                                >
                                  <mock-icon
                                    aria-hidden=""
                                    class="MuiSvgIcon-root MuiSvgIcon-colorBorder MuiSvgIcon-fontSizeSmall css-gvpe62-MuiSvgIcon-root"
                                    data-testid="copy-btn-icon"
                                    focusable="false"
                                  />
                                </button>
                              </span>
                              0x
                               
                            </div>
                          </div>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-container css-86qz5o-MuiGrid-root"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item css-1wehkfy-MuiGrid-root"
                            data-testid="tx-row-title"
                            style="word-break: break-word;"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-shf88x-MuiTypography-root"
                            >
                              operation:
                            </p>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
                            data-testid="tx-data-row"
                          >
                            0 (call)
                          </div>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-container css-86qz5o-MuiGrid-root"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item css-1wehkfy-MuiGrid-root"
                            data-testid="tx-row-title"
                            style="word-break: break-word;"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-shf88x-MuiTypography-root"
                            >
                              nonce:
                            </p>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
                            data-testid="tx-data-row"
                          >
                            100
                          </div>
                        </div>
                        <div
                          class="MuiBox-root css-kz5wc9"
                        />
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-111k8jw-MuiTypography-root"
                        >
                          Transaction hashes
                        </p>
                        <div
                          class="MuiStack-root css-hwnj0i-MuiStack-root"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-container css-86qz5o-MuiGrid-root"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-item css-1wehkfy-MuiGrid-root"
                              data-testid="tx-row-title"
                              style="word-break: break-word;"
                            >
                              <p
                                class="MuiTypography-root MuiTypography-body1 css-shf88x-MuiTypography-root"
                              >
                                Domain hash:
                              </p>
                            </div>
                            <div
                              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
                              data-testid="tx-data-row"
                            >
                              <div
                                class="encodedData MuiBox-root css-0"
                                data-testid="tx-hexData"
                              >
                                <span
                                  aria-label="Copy to clipboard"
                                  class=""
                                  data-mui-internal-clone-element="true"
                                  style="cursor: pointer;"
                                >
                                  <button
                                    aria-label="Copy to clipboard"
                                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-xyrzjn-MuiButtonBase-root-MuiIconButton-root"
                                    tabindex="0"
                                    type="button"
                                  >
                                    <mock-icon
                                      aria-hidden=""
                                      class="MuiSvgIcon-root MuiSvgIcon-colorBorder MuiSvgIcon-fontSizeSmall css-gvpe62-MuiSvgIcon-root"
                                      data-testid="copy-btn-icon"
                                      focusable="false"
                                    />
                                  </button>
                                </span>
                                 
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1w4z6qv-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: none;"
  >
    <div
      class="MuiCardContent-root cardContent css-1lt5qva-MuiCardContent-root"
      data-testid="card-content"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1whrnku-MuiTypography-root"
      >
        Transaction checks
      </h5>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorInfo MuiAlert-standardInfo MuiAlert-standard css-c7us71-MuiPaper-root-MuiAlert-root"
        role="alert"
        style="--Paper-shadow: none;"
      >
        <div
          class="MuiAlert-icon css-vab54s-MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-zysrwz-MuiSvgIcon-root"
            data-testid="InfoOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message css-zioonp-MuiAlert-message"
        >
          We recommend analyzing your transaction with a third-party tool like
           
          <span
            data-track="modals: Open Safe Utils"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-r9pq5a-MuiTypography-root-MuiLink-root"
              href="https://safeutils.openzeppelin.com/"
              rel="noreferrer noopener"
              target="_blank"
            >
              <span
                class="MuiBox-root css-1vqf8mi"
              >
                Safe Utils
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-tqxw8e-MuiSvgIcon-root"
                  data-testid="OpenInNewRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 19H6c-.55 0-1-.45-1-1V6c0-.55.45-1 1-1h5c.55 0 1-.45 1-1s-.45-1-1-1H5c-1.11 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-6c0-.55-.45-1-1-1s-1 .45-1 1v5c0 .55-.45 1-1 1M14 4c0 .55.45 1 1 1h2.59l-9.13 9.13c-.39.39-.39 1.02 0 1.41s1.02.39 1.41 0L19 6.41V9c0 .55.45 1 1 1s1-.45 1-1V4c0-.55-.45-1-1-1h-5c-.55 0-1 .45-1 1"
                  />
                </svg>
              </span>
            </a>
          </span>
          . Third-party terms & disclaimers may apply.
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1w4z6qv-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: none;"
  >
    <div
      class="MuiCardContent-root cardContent css-1lt5qva-MuiCardContent-root"
      data-testid="card-content"
    >
      <div
        class="wrapper"
      >
        <div
          class="icon sign"
        >
          <mock-icon
            aria-hidden=""
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-tqxw8e-MuiSvgIcon-root"
            focusable="false"
          />
        </div>
        <div>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1mp5g96-MuiTypography-root"
          >
            confirm
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body2 css-17vdyq3-MuiTypography-root"
          >
            You're about to 
            create and 
            confirm
             this transaction.
          </p>
        </div>
      </div>
      <form>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth nestedDivider css-1jpvyob-MuiDivider-root"
        />
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1q4nm6f-MuiCardActions-root"
        >
          <div
            class="MuiStack-root css-irtfmw-MuiStack-root"
          >
            <span
              class=""
              data-mui-internal-clone-element="true"
            >
              <span
                data-track="batching: Add to batch"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-16be8yl-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  <mock-icon
                    aria-hidden=""
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1hr8a3z-MuiSvgIcon-root"
                    focusable="false"
                  />
                  Add to batch
                </button>
              </span>
            </span>
            <div
              class="MuiBox-root css-1aqmpzz"
            >
               
              <div
                aria-orientation="horizontal"
                class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1lftdsp-MuiDivider-root"
                role="separator"
              >
                <span
                  class="MuiDivider-wrapper css-tuok6n-MuiDivider-wrapper"
                >
                  or
                </span>
              </div>
            </div>
            <span
              aria-label="Please connect your wallet"
              class=""
              data-mui-internal-clone-element="true"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1hp2ejb-MuiButtonBase-root-MuiButton-root"
                data-testid="sign-btn"
                disabled=""
                tabindex="-1"
                type="submit"
              >
                Sign
              </button>
            </span>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`SignOrExecute should display a loading component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1w4z6qv-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: none;"
  >
    <div
      class="MuiCardContent-root cardContent css-1lt5qva-MuiCardContent-root"
      data-testid="card-content"
    >
      <div
        class="MuiBox-root css-17luq05"
      >
        <div
          class="box MuiBox-root css-0"
        >
          <div
            class="rect rectTl"
          />
          <div
            class="rect rectTr"
          />
          <div
            class="rect rectBl"
          />
          <div
            class="rect rectBr"
          />
          <div
            class="rect rectCenter"
          />
          <svg
            version="1.1"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <filter
                id="gooey"
              >
                <fegaussianblur
                  in="SourceGraphic"
                  result="blur"
                  stdDeviation="3"
                />
                <fecolormatrix
                  in="blur"
                  mode="matrix"
                  result="goo"
                  values="1 0 0 0 0  0 1 0 0 0  0 0 1 0 0  0 0 0 19 -9"
                />
                <fecomposite
                  in="SourceGraphic"
                  in2="goo"
                  operator="atop"
                />
              </filter>
            </defs>
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SignOrExecute should display an error screen 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1w4z6qv-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: none;"
  >
    <div
      class="MuiCardContent-root cardContent css-1lt5qva-MuiCardContent-root"
      data-testid="card-content"
    >
      <div
        class="MuiStack-root css-1sazv7p-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1gdgqta-MuiPaper-root-MuiAccordion-root"
            style="--Paper-shadow: none;"
          >
            <h3
              class="MuiAccordion-heading css-cy7rkm-MuiAccordion-heading"
            >
              <button
                aria-expanded="false"
                class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters accordion css-10sjung-MuiButtonBase-root-MuiAccordionSummary-root"
                data-testid="decoded-tx-summary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-1r0e0ir-MuiAccordionSummary-content"
                >
                  Advanced details
                  <span
                    class=""
                    data-mui-internal-clone-element="true"
                  >
                    <mock-icon
                      aria-hidden=""
                      class="MuiSvgIcon-root MuiSvgIcon-colorBorder MuiSvgIcon-fontSizeSmall css-17ibv3e-MuiSvgIcon-root"
                      focusable="false"
                    />
                  </span>
                  <div
                    class="MuiBox-root css-1rr4qq7"
                  />
                </span>
                <span
                  class="MuiAccordionSummary-expandIconWrapper css-1wqf3nl-MuiAccordionSummary-expandIconWrapper"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1dhtbeh-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </span>
              </button>
            </h3>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-cwrbtg-MuiCollapse-root"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
                >
                  <div
                    class="MuiAccordion-region"
                    role="region"
                  >
                    <div
                      class="MuiAccordionDetails-root css-w74p4c-MuiAccordionDetails-root"
                      data-testid="decoded-tx-details"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1w4z6qv-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: none;"
  >
    <div
      class="MuiCardContent-root cardContent css-1lt5qva-MuiCardContent-root"
      data-testid="card-content"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1whrnku-MuiTypography-root"
      >
        Transaction checks
      </h5>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorInfo MuiAlert-standardInfo MuiAlert-standard css-c7us71-MuiPaper-root-MuiAlert-root"
        role="alert"
        style="--Paper-shadow: none;"
      >
        <div
          class="MuiAlert-icon css-vab54s-MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-zysrwz-MuiSvgIcon-root"
            data-testid="InfoOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message css-zioonp-MuiAlert-message"
        >
          We recommend analyzing your transaction with a third-party tool like
           
          <span
            data-track="modals: Open Safe Utils"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-r9pq5a-MuiTypography-root-MuiLink-root"
              href="https://safeutils.openzeppelin.com/"
              rel="noreferrer noopener"
              target="_blank"
            >
              <span
                class="MuiBox-root css-1vqf8mi"
              >
                Safe Utils
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-tqxw8e-MuiSvgIcon-root"
                  data-testid="OpenInNewRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 19H6c-.55 0-1-.45-1-1V6c0-.55.45-1 1-1h5c.55 0 1-.45 1-1s-.45-1-1-1H5c-1.11 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-6c0-.55-.45-1-1-1s-1 .45-1 1v5c0 .55-.45 1-1 1M14 4c0 .55.45 1 1 1h2.59l-9.13 9.13c-.39.39-.39 1.02 0 1.41s1.02.39 1.41 0L19 6.41V9c0 .55.45 1 1 1s1-.45 1-1V4c0-.55-.45-1-1-1h-5c-.55 0-1 .45-1 1"
                  />
                </svg>
              </span>
            </a>
          </span>
          . Third-party terms & disclaimers may apply.
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1w4z6qv-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: none;"
  >
    <div
      class="MuiCardContent-root cardContent css-1lt5qva-MuiCardContent-root"
      data-testid="card-content"
    >
      <div
        class="wrapper"
      >
        <div
          class="icon sign"
        >
          <mock-icon
            aria-hidden=""
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-tqxw8e-MuiSvgIcon-root"
            focusable="false"
          />
        </div>
        <div>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1mp5g96-MuiTypography-root"
          >
            confirm
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body2 css-17vdyq3-MuiTypography-root"
          >
            You're about to 
            create and 
            confirm
             this transaction.
          </p>
        </div>
      </div>
      <form>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth nestedDivider css-1jpvyob-MuiDivider-root"
        />
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1q4nm6f-MuiCardActions-root"
        >
          <div
            class="MuiStack-root css-irtfmw-MuiStack-root"
          >
            <span
              class=""
              data-mui-internal-clone-element="true"
            >
              <span
                data-track="batching: Add to batch"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-16be8yl-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  <mock-icon
                    aria-hidden=""
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1hr8a3z-MuiSvgIcon-root"
                    focusable="false"
                  />
                  Add to batch
                </button>
              </span>
            </span>
            <div
              class="MuiBox-root css-1aqmpzz"
            >
               
              <div
                aria-orientation="horizontal"
                class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1lftdsp-MuiDivider-root"
                role="separator"
              >
                <span
                  class="MuiDivider-wrapper css-tuok6n-MuiDivider-wrapper"
                >
                  or
                </span>
              </div>
            </div>
            <span
              aria-label="Please connect your wallet"
              class=""
              data-mui-internal-clone-element="true"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1hp2ejb-MuiButtonBase-root-MuiButton-root"
                data-testid="sign-btn"
                disabled=""
                tabindex="-1"
                type="submit"
              >
                Sign
              </button>
            </span>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
